src/components/WebAnimationDemo/useAnimation.tsx

```diff
@@ -0,0 +1,53 @@
+import { useCallback, useEffect, useRef } from "react";
+
+export const useAnimation = (
+  keyframes: Keyframe[] | PropertyIndexedKeyframes,
+  options?: KeyframeAnimationOptions
+): {
+  ref: React.MutableRefObject<HTMLDivElement>;
+  animation: Animation;
+  pause: () => void;
```

Comment:
I think I would prefer using the animation object directly instead of just replicating its APIs here. The only exception seem to be position, which we could have as a helper function taking an Animation object as a parameter maybe to make it more reusable.

Reply:
Sounds good.